pkg/credentials/credentials.go

```diff
@@ -0,0 +1,127 @@
+package credentials
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"gopkg.in/yaml.v2"
+)
+
+// ReadFile is used to extract user/pass credentials for connecting to ES from a file using the provided user/pass
+// extraction paths.
+func ReadFile(fileName, tplUser, tplPass string) (user string, pass string, err error) {
+	var (
+		b   []byte
+		obj interface{}
+	)
+
+	b, err = ioutil.ReadFile(fileName)
+	if err != nil {
+		return "", "", err
+	}
+
+	switch strings.ToLower(filepath.Ext(fileName)) {
+	case ".yml", ".yaml":
```

Comment:
Why should we support multiple file types since vault agent can generate any format we want?

Reply:
Because this is OSS and both preferences and applications differ.